.github/workflows/generate-and-publish-docs.yml

```diff
@@ -0,0 +1,29 @@
+name: Generate and Deploy Doxygen Docs
+
+on:
+  push:
+    tags:
+      - '*' # Only re-generate docs when a new version is pushed
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v5
+
+      - name: Install Doxygen and Graphviz
+        run: sudo apt-get install -y doxygen graphviz
+
+      - name: Generate Documentation
+        run: doxygen Doxyfile
+
+      - name: Deploy to cockatrice.github.io
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          deploy_key: ${{ secrets.DOCS_DEPLOY_KEY }}
+          external_repository: Cockatrice/cockatrice.github.io
+          publish_branch: main
```

Comment:
just noticed a rather annoying flaw, our main branch on the website repo is actually master